R/check_crpr.R

```diff
@@ -0,0 +1,131 @@
+.datasets <- new.env(parent = emptyenv())
+
+#' Signal CR Records Followed by PR
+#'
+#' @param dataset A data frame
+#'
+#' @param order A list of variables created using `vars()` determining the order
+#'   or the records
+#'
+#' @param msg The condition message
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of symbols created using `vars()` is expected.
+#'
+#' @param check_type Type of message to issue when detecting PR after CR.
+#'
+#'   *Default*: `"warning"`
+#'
+#'   *Permitted Values*: `"warning"` or `"error"`
+#'
+#' @return No return value, called for side effects
+#'
+#' @seealso [get_crpr_dataset()]
+#'
+#' @export
+#' @keywords user_utility
+#' @author Stefan Bundfuss
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr)
+#' library(lubridate)
+#' library(admiralonco)
+#'
+#' adrs <- tribble(
+#'   ~USUBJID, ~ADTC,        ~AVALC,
+#'   "1",      "2020-01-01", "PR",
+#'   "1",      "2020-02-01", "CR",
+#'   "1",      "2020-02-16", "NE",
+#'   "1",      "2020-03-01", "CR",
+#'   "2",      "2020-02-06", "PR",
+#'   "2",      "2020-02-16", "CR",
+#'   "2",      "2020-03-30", "PR",
+#' ) %>%
+#'   mutate(
+#'     ADT = ymd(ADTC),
+#'     STUDYID = "XX1234"
+#'   )
+#'
+#' signal_crpr(adrs, order = vars(ADT))
+signal_crpr <- function(dataset,
+                        order,
+                        msg = "Dataset contains CR records followed by PR.",
+                        subject_keys = vars(STUDYID, USUBJID),
+                        check_type = "warning") {
+  assert_order_vars(order)
+  assert_vars(subject_keys)
+  assert_data_frame(dataset, required_vars = quo_c(order, subject_keys))
+  assert_character_scalar(check_type, values = c("warning", "error"))
+
+  crpr_data <- filter_confirmation(
+    dataset,
+    by_vars = subject_keys,
+    order = order,
+    join_vars = vars(AVALC),
+    join_type = "after",
+    filter = AVALC == "CR" & AVALC.join == "PR"
+  )
+
+  if (nrow(crpr_data) > 0) {
+    .datasets$crpr <- crpr_data
+    full_msg <- paste0(
+      msg,
+      "\nRun `get_crpr_dataset()` to access the CR records records followed by PR"
+    )
+    msg_funs <- list(warning = warn, error = abort)
+    msg_funs[[check_type]](full_msg)
+  }
+}
+
+#' Get CR Records Followed by PR That Lead to a Prior Error
+#'
+#' @export
+#'
+#' @author Stefan Bundfuss
+#'
+#' @details
+#' Some {admiralonco} function check that the in the source records CR is not
```

Comment:
Change first part to: Some tumor response functions looking at confirmation of response check that any source records Complete Response (CR) is not followed by Partial Response (PR)...

As makes more robust when these move to admiraldev